documentation/src/main/asciidoc/core/manual/en_US/jpql_functions.adoc

```diff
@@ -214,6 +214,32 @@ This can be used for cases when the type of an expression is actually known but
 
 WARNING: This is an internal function that is used to implement the <<values-clause,`VALUES` clause>> for _basic_ types. It is not intended for direct use and might change without notice.
 
+==== LITERAL functions
+
+Syntax: `LITERAL_XXX ( argument )`
+
+WARNING: This function is used internally and no user should ever have the need for this!
+
+There are multiple different literal functions for different data types.
+
+* `java.util.Calendar` - `LITERAL_CALENDAR`
+* `java.sql.Time` - `LITERAL_TIME`
+* `java.sql.Date` - `LITERAL_DATE`
+* `java.sql.Timestamp` - `LITERAL_TIMESTAMP`
+* `java.util.Date` - `LITERAL_UTIL_DATE`
+* `java.time.Instant` - `LITERAL_INSTANT`
+* `java.time.LocalDate` - `LITERAL_LOCAL_DATE`
```

Comment:
Don't we want a `TREAT_LOCAL_DATE` following the same mechanism as well?

Reply:
Thanks for the hint. We already have that and I now added the documentation for this.